slice-core-api/src/main/java/com/cognifide/slice/api/tag/SliceLookupTag.java

```diff
@@ -49,14 +49,17 @@ public void doTag() throws JspException {
 			}
 
 			final PageContext pageContext = (PageContext) getJspContext();
-			final Object model = SliceTagUtils.getFromCurrentPath(pageContext, type, appName);
+			final Class typeClass = SliceTagUtils.getClassFromType(pageContext, type);
+			final Object model = SliceTagUtils.getFromCurrentPath(pageContext, typeClass, appName);
 			pageContext.setAttribute(var, model, PageContext.PAGE_SCOPE);
+		} catch (ClassNotFoundException e) {
+			throw new JspTagException("Could not get class for " + type);
 		} finally {
 			clean();
 		}
 	}
 
-	public void setType(Class<?> type) {
+	public void setType(String type) {

```

Comment:
I'd double check the overloading - the blog post you referenced is mentioning JavaBeans not taglibs.